src/main/java/com/cloudbees/jenkins/support/util/Anonymizer.java

```diff
@@ -0,0 +1,207 @@
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.ConcurrentSkipListSet;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@Restricted(NoExternalUse.class)
+public class Anonymizer {
+    private static final Logger LOGGER = Logger.getLogger(Anonymizer.class.getName());
+    private static final RandomNameGenerator GENERATOR = new RandomNameGenerator();
+    private static final Set<String> SEPARATORS = new CopyOnWriteArraySet<>();
+    private static final Map<String, String> ANON_MAP = new ConcurrentHashMap<>();
+    private static final Set<String> ORDER = new ConcurrentSkipListSet<>(Comparator.comparingInt(String::length).reversed().thenComparing(s -> s));
+    private static final Map<String, String> DISPLAY = new ConcurrentSkipListMap<>();
+    private static final long DISPLAY_REFRESH_INTERVAL = 1000 * 60 * 10; // 10 mins
+
+    private static long LAST_REFRESH;
+
+    private Anonymizer() { }
+
+    static {
+        // Full names can have this separator
+        SEPARATORS.add(" » ");
+        refresh();
+    }
+
+    @SuppressWarnings("unchecked")
+    public static void refresh() {
+        LOGGER.log(Level.FINE, "Refreshing anonymized items");
+        Jenkins instance = Jenkins.getInstance();
+        load();
+
+        SupportPlugin.AnonymizationSettings settings = SupportPlugin.getInstance().getAnonymizationSettings();
+        if (settings.isAnonymizeLabels()) {
+            for (Label label : instance.getLabels()) {
+                anonymizeName(label.getName(), "label", "", false);
+            }
+        }
+
+        if (settings.isAnonymizeItems()) {
+            for (Item item : instance.getAllItems()) {
+                anonymizePath(item.getFullName(), "item", false);
+            }
+        }
+
+        if (settings.isAnonymizeViews()) {
+            for (View view : instance.getViews()) {
+                anonymizeName(view.getViewName(), "view", "", false);
+            }
+        }
+
+        if (settings.isAnonymizeNodes()) {
+            for (Node node : instance.getNodes()) {
+                anonymizeName(node.getDisplayName(), "node", "", false);
+            }
+        }
+
+        if (settings.isAnonymizeComputers()) {
+            for (Computer computer : instance.getComputers()) {
+                anonymizeName(computer.getDisplayName(), "computer", "", false);
+            }
+        }
+
+        if (settings.isAnonymizeUsers()) {
+            for (User user : User.getAll()) {
+                anonymizeName(user.getDisplayName(), "user", "", false);
+            }
+        }
+
+        save();
+    }
+
+    public static String anonymize(String input) {
+        String anonymized = input;
+        for (String key : ORDER) {
+            anonymized = anonymized.replaceAll("(?i)\\b(" + key + ")\\b", ANON_MAP.get(key));
```

Comment:
We need to escape `key` here so that Regex control characters are treated as literals. Right now having a Job with a weird name such as `J1-test_test2{hmm}+nice` throws a `PatternSyntaxException` in `replaceAll`.

Reply:
TIL: https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#quote-java.lang.String-